src/qml/qmlplayerproxy.cpp

```diff
@@ -67,124 +44,40 @@ void QmlPlayerProxy::loadTrackFromLocationUrl(const QUrl& trackLocationUrl, bool
     }
 }
 
-void QmlPlayerProxy::slotTrackLoaded(TrackPointer pTrack) {
-    m_pCurrentTrack = pTrack;
-    if (pTrack != nullptr) {
-        connect(pTrack.get(),
-                &Track::artistChanged,
-                this,
-                &QmlPlayerProxy::artistChanged);
-        connect(pTrack.get(),
-                &Track::titleChanged,
-                this,
-                &QmlPlayerProxy::titleChanged);
-        connect(pTrack.get(),
-                &Track::albumChanged,
-                this,
-                &QmlPlayerProxy::albumChanged);
-        connect(pTrack.get(),
-                &Track::albumArtistChanged,
-                this,
-                &QmlPlayerProxy::albumArtistChanged);
-        connect(pTrack.get(),
-                &Track::genreChanged,
-                this,
-                &QmlPlayerProxy::genreChanged);
-        connect(pTrack.get(),
-                &Track::composerChanged,
-                this,
-                &QmlPlayerProxy::composerChanged);
-        connect(pTrack.get(),
-                &Track::groupingChanged,
-                this,
-                &QmlPlayerProxy::groupingChanged);
-        connect(pTrack.get(),
-                &Track::yearChanged,
-                this,
-                &QmlPlayerProxy::yearChanged);
-        connect(pTrack.get(),
-                &Track::trackNumberChanged,
-                this,
-                &QmlPlayerProxy::trackNumberChanged);
-        connect(pTrack.get(),
-                &Track::trackTotalChanged,
-                this,
-                &QmlPlayerProxy::trackTotalChanged);
-        connect(pTrack.get(),
-                &Track::commentChanged,
-                this,
-                &QmlPlayerProxy::commentChanged);
-        connect(pTrack.get(),
-                &Track::keyChanged,
-                this,
-                &QmlPlayerProxy::keyTextChanged);
-        connect(pTrack.get(),
-                &Track::colorUpdated,
-                this,
-                &QmlPlayerProxy::colorChanged);
-        connect(pTrack.get(),
-                &Track::waveformUpdated,
-                this,
-                &QmlPlayerProxy::slotWaveformChanged);
-        connect(pTrack.get(),
-                &Track::beatsUpdated,
-                this,
-                &QmlPlayerProxy::slotBeatsChanged);
-        connect(pTrack.get(),
-                &Track::cuesUpdated,
-                this,
-                &QmlPlayerProxy::slotHotcuesChanged);
-#ifdef __STEM__
-        connect(pTrack.get(),
-                &Track::stemsUpdated,
-                this,
-                &QmlPlayerProxy::slotStemsChanged);
-#endif
-        slotBeatsChanged();
-        slotHotcuesChanged();
+void QmlPlayerProxy::subclassesSlotTrackLoaded() {
```

Comment:
I believe this is a bit of an anti pattern. I think it would be best to let sub classes register a connection directly, in case they need custom actions